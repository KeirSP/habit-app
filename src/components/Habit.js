import React, { Component } from 'react';

class Habit extends Component {
    constructor(props) {
        super(props);
        this.state={
            data:this.props.data,
            today: false
        }
    }


    doneToday = () => {
        return this.setState({today: !this.state.today})
    }
    
    render() {
        return (
            <div>
                <h3>{this.state.data.activity}</h3>
                <p>{this.state.data.streak}</p>
                <label>
                    <input type="checkbox" value={this.state.today} name="today" onChange={this.doneToday} />
                    Have you done this today?
                </label>
            </div>
        );
    }
}

export default Habit;