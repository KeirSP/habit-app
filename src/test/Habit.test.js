import React from 'react';
import ReactDOM from 'react-dom'
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Habit from '../components/Habit'

describe(Habit, () => {
    // it('mounts to DOM', () => {
    //     const div = document.createElement('div');
    //     ReactDOM.render(<Habit />, div);
    //     ReactDOM.unmountComponentAtNode(div);
    // });

    // it('renders correctly', () => {
    //     const tree = renderer
    //         .create(<Habit />)
    //         .toJSON();
    //     expect(tree).toMatchSnapshot();
    // });

    let wrapper;
    beforeEach(() => wrapper = shallow(<Habit />));
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    // it('should render a <div />', () => {
    //     expect(wrapper.find('div').length).toEqual(1);
    // });

    // it('should render a <button />', () => {
    //     expect(wrapper.find('button').length).toEqual(1);
    // });

    // it('Hbit check with right data',()=>{
    //     wrapper = shallow(<Habit/>);
    //     wrapper.find('input[type="text"]').simulate('change', {target: {name: 'today', value: 'false'}});
    //     wrapper.find('button').simulate('click');
    //     expect(wrapper.state('today')).toBe(true);
    // })


})