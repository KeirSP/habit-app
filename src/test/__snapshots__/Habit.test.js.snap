// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habit Renders correctly 1`] = `undefined`;
